crates/nao/src/lib.rs

```diff
@@ -333,3 +359,33 @@ impl Display for Network {
         }
     }
 }
+
+fn extract_version_num(input:&str) -> Option<String>{
+    let lines = input.lines();
+        for line in lines {
+            if line.contains("VERSION_ID") {
+                let Some((_, os_version)) = line.split_once('=') else { continue; };
+		return Some(os_version.to_string());
+            }
+        }
+   None
+}    
+#[cfg(test)]
```

Comment:
I noticed a lot of whitespace errors in your PR. Around functions, impls, structs, enums we want to have one empty line. Within curly braces we want no empty lines in the beginning and end of the containing items.

Also, your code is not formatted. We are using automatic formatters for this. Either press Ctrl+Shift+I in your VS Code or run `cargo fmt` on the command line. Beware that the formatting does not fix empty line mistakes.